TaskModules/powershell/Azure/SPNCreation.ps1

```diff
@@ -1,7 +1,7 @@
 param
 (
-    [Parameter(Mandatory=$true, HelpMessage="Enter Azure Subscription name. You need to be Subscription Admin to execute the script")]
-    [string] $subscriptionName,
+    [Parameter(Mandatory=$false, HelpMessage="Enter Scope Level for endpoint. Scope Level can either be Management Group or Subscription")]

```

Comment:
Instead of this, can you use parameter set. You can have one parameter set based on subscription id and the other based on management group id. Keep subscription as the default parameter set.
https://social.technet.microsoft.com/wiki/contents/articles/15994.powershell-advanced-function-parameter-attributes.aspx#ParameterSetName